Modules/Office365DSC/DSCResources/MSFT_SCRetentionComplianceRule/MSFT_SCRetentionComplianceRule.psm1

```diff
@@ -0,0 +1,283 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Policy,
+
+        [Parameter()]
+        [System.String]
+        $Comment,
+
+        [Parameter()]
+        [System.String[]]
+        $ExcludedItemClasses,
+
+        [Parameter()]
+        [System.String]
+        $RetentionDuration,
+
+        [Parameter()]
+        [ValidateSet("Days", "Months", "Years")]
+        [System.String]
+        $RetentionDurationDisplayHint,
+
+        [Parameter()]
+        [System.String]
+        $ContentMatchQuery,
+
+        [Parameter()]
+        [ValidateSet("CreationAgeInDays", "ModificationAgeInDays")]
+        [System.String]
+        $ExpirationDateOption,
+
+        [Parameter()]
+        [ValidateSet("Delete","Keep","KeepAndDelete")]
+        [System.String]
+        $RetentionComplianceAction,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Getting configuration of RetentionComplianceRule for $Name"
+
+    Write-Verbose -Message "Calling Test-SecurityAndComplianceConnection function:"
+    Test-SecurityAndComplianceConnection -GlobalAdminAccount $GlobalAdminAccount
+
+    $RuleObjects = Get-RetentionComplianceRule
+    $RuleObject = $RuleObjects | Where-Object {$_.Identity -eq $Name}
+
+    if ($null -eq $RuleObject)
+    {
+        Write-Verbose -Message "RetentionComplianceRule $($Name) does not exist."
+        $result = $PSBoundParameters
+        $result.Ensure = 'Absent'
+        return $result
+    }
+    else
+    {
+        Write-Verbose "Found existing RetentionComplianceRule $($Name)"
+        $result = @{
+            Ensure = 'Present'
+        }
+        foreach ($KeyName in ($PSBoundParameters.Keys | Where-Object -FilterScript { $_ -ne 'Ensure' }))
+        {
+            if ($null -ne $RuleObject.$KeyName)
+            {
+                $result += @{
+                    $KeyName = $RuleObject.$KeyName
+                }
+            }
+            else
+            {
+                $result += @{
+                    $KeyName = $PSBoundParameters[$KeyName]
+                }
+            }
+        }
+
+        Write-Verbose -Message "Found RetentionComplianceRule $($Name)"
+        Write-Verbose -Message "Get-TargetResource Result: `n $(Convert-O365DscHashtableToString -Hashtable $result)"
```

Comment:
The Test method also outputs this info. Since that also calls the Get method, this info is outputted to the screen twice. Might be good to call this just once.

Reply:
Kept it only in the Get, this way we also target the ReverseDSC scenario.